packages/mobx-state-tree/src/core/node/BaseNode.ts

```diff
@@ -15,9 +15,23 @@ import { createAtom } from "mobx"
 export abstract class BaseNode {
     protected readonly subpathAtom = createAtom(`path`)
     protected escapedSubpath: string
-    state = NodeLifeCycle.INITIALIZING
     storedValue: any
 
+    private readonly aliveAtom = createAtom(`alive`)
+    private _state = NodeLifeCycle.INITIALIZING
+    get state() {
+        return this._state
+    }
+    set state(val: NodeLifeCycle) {
+        const wasAlive = this.isAlive
+        this._state = val
+        const isAlive = this.isAlive
+
+        if (wasAlive !== isAlive) {
+            this.aliveAtom.reportChanged()
```

Comment:
I'm not sure if it's safe to call ```reportChanged()``` if you did not ```reportObserved()``` before?
I mean if you had not used ```observableIsAlive``` and node became ```DEAD``` so we hit this code: wouldnt it accidantely throw? It's just a suggestion, as I am not aquainted with mobx internals deep enough.

Reply:
I think it is ok (or else a gazillion unit tests would throw) but I'll change it so the atom is only created when the observable is alive is used and that way we kill two birds with one stone

Reply:
`reportChanged()` and `reportObserved()` are allowed to be called without any further assumptions on the other to be called